components/plan-build/bin/hab-plan-build.sh

```diff
@@ -953,30 +946,27 @@ _resolve_scaffolding_dependencies() {
   scaff_build_deps=()
   scaff_build_deps_resolved=()
 
-  # shellcheck disable=2066
-  for dep in "${pkg_scaffolding}"; do
-    _install_dependency "$dep"
-    # Add scaffolding package to the list of scaffolding build deps
-    scaff_build_deps+=($dep)
-    if resolved="$(_resolve_dependency "$dep")"; then
-      build_line "Resolved scaffolding dependency '$dep' to $resolved"
-      scaff_build_deps_resolved+=($resolved)
-      # Add each (fully qualified) direct run dependency of the scaffolding
-      # package.
-      sdeps=($(_get_deps_for "$resolved"))
-      for sdep in "${sdeps[@]}"; do
-        scaff_build_deps+=($sdep)
-        scaff_build_deps_resolved+=($HAB_PKG_PATH/$sdep)
-      done
-    else
-      exit_with "Resolving '$dep' failed, should this be built first?" 1
-    fi
-  done
+  _install_dependency "$pkg_scaffolding"
+  # Add scaffolding package to the list of scaffolding build deps
+  scaff_build_deps+=("$pkg_scaffolding")
+  if resolved="$(_resolve_dependency "$pkg_scaffolding")"; then
+    build_line "Resolved scaffolding dependency '$pkg_scaffolding' to $resolved"
+    scaff_build_deps_resolved+=("$resolved")
+    # Add each (fully qualified) direct run dependency of the scaffolding
+    # package.
+    mapfile -t sdeps < <(_get_deps_for "$resolved")
```

Comment:
@baumanj Could you provide a brief comment inline describing what the `< <()` syntax means? I'm having a little trouble understanding it.  I see in your pr you link to the shellcheck page that describes that this is the preferred style, but didn't see an explanation of the syntax.  Is it a particular bash idiom we could link to?

Reply:
Added https://github.com/habitat-sh/habitat/pull/6257/commits/d60e5212754f6c870429eb0786baa700a7ddce8e